src/gtfs/read.rs

```diff
@@ -744,10 +750,11 @@ fn get_modes_from_gtfs(
         .iter()
         .map(|mt| get_commercial_mode(mt))
         .collect();
-    let physical_modes = gtfs_mode_types
+    let physical_modes_set: HashSet<objects::PhysicalMode> = gtfs_mode_types
```

Comment:
I think you should remove the explicit type

Reply:
it has to be there otherwise it gets a vector, we use hashset first to remove duplicate and then transform to a vector for function return